msal-dotnet-articles/acquiring-tokens/desktop-mobile/wam.md

```diff
@@ -1,63 +1,87 @@
 
-After referencing the correct packages, call `WithBroker(BrokerOptions)` with options and pass a window handle.
+After referencing the relevant packages, call [`WithBroker(BrokerOptions)`](xref:Microsoft.Identity.Client.Desktop.WamExtension.WithBroker*) with broker configuration options and [a window handle](#parent-window-handles) that the broker will be bound to.
 
 ```csharp
-var pca = PublicClientApplicationBuilder.Create(CLIENT_ID)
-            .WithAuthority("https://login.microsoftonline.com/common")
-            .WithBroker(new BrokerOptions(BrokerOptions.OperatingSystems.Windows))
-            .WithParentActivityOrWindow(() => WINDOW_HANDLE)
-            .Build();
+BrokerOptions options = new BrokerOptions(BrokerOptions.OperatingSystems.Windows);
+
+options.Title = "My Awesome Application";
+options.ListOperatingSystemAccounts = true;
+
+IPublicClientApplication app =
+    PublicClientApplicationBuilder.Create("YOUR_CLIENT_ID")
+    .WithDefaultRedirectUri()
+    .WithParentActivityOrWindow(GetConsoleOrTerminalWindow)
+    .WithBroker(options)
+    .Build();
+
+var authResult = await app.AcquireTokenInteractive(new List<string>() { "User.Read" }).ExecuteAsync();
 ```
 
-No changes are required for UWP applications. Because the platform does not support the updated broker, existing applications will continue to use the legacy WAM implementation, which is documented in [Acquire a token using WAM](/azure/active-directory/develop/scenario-desktop-acquire-token-wam).
+When using the broker, if the [authority](/azure/active-directory/develop/msal-client-application-configuration#authority) used is targeting Azure AD as well as personal Microsoft accounts, the user will first be prompted to select an account using the built-in system account picker.
+
+![Demo of the WAM component](../../media/wam/wam-demo.gif)
+
+If the configuration is set on a per-tenant basis by using [`WithTenantId`](xref:Microsoft.Identity.Client.AbstractApplicationBuilder`1.WithTenantId*) or if the authority is set to an [audience](xref:Microsoft.Identity.Client.AadAuthorityAudience) that _does not_ include personal Microsoft accounts, the native Windows account picker will not be shown and instead the user will be prompted with a generic Microsoft authentication prompt.
+
+![Demo of the WAM component that is configured on a per-tenant basis and doesn't show the OS-based account picker](../../media/wam/wam-per-tenant.gif)
+
+Once the account is added or selected, the user will be prompted for additional consent if they have never used the application before or the application requires additional permissions.
+
+>[!NOTE]
+>No changes are required for UWP applications. Because the platform does not support the updated broker, existing applications will continue to use the legacy WAM implementation, which is documented in [Acquire a token using WAM](/azure/active-directory/develop/scenario-desktop-acquire-token-wam).
```

Comment:
```suggestion
>No changes are required for UWP applications. Because the platform does not support the updated broker, existing applications will continue to use the legacy WAM implementation, which is documented in [Acquire a token using WAM](/azure/active-directory/develop/scenario-desktop-acquire-token-wam).
```
the link still leads to a page that talks about the new broker

Reply:
At some point we need to merge that page into the current one. I removed the reference for now.